### ISSUE TITLE
about 43% faster on CPUs

### ISSUE THREAD

**Issue by tongchangD** [ultralytics/ultralytics#24304]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

The official website mentioned：
Together, these innovations deliver a model family that achieves higher accuracy on small objects, provides seamless deployment,
and runs up to 43% faster on CPUs — making YOLO26 one of the most practical and deployable YOLO models to date for resource-constrained environments.
My environment：
```
-> uname -a
Linux 5.15.77-amd64-desktop
-> nvcc -V 
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Tue_May__3_18:49:52_PDT_2022
Cuda compilation tools, release 11.7, V11.7.64
Build cuda_11.7.r11.7/compiler.31294372_0
-> nvidia-smi
NVIDIA GeForce RTX 3070
-> cat /proc/cpuinfo | grep "model name" | head -1
model name      : Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
-> pip list 
Package                  Version
------------------------ -----------
torch                    2.0.1
torchaudio               2.0.2
torchvision              0.15.2
ultralytics              8.3.48
ultralytics-thop         2.0.13

-> python --version
Python 3.10.6

```
What is the basis for the reported 43% optimization?
Preliminary experimental results indicate that YOLO26 is inferior to YOLO11 in both detection accuracy and robustness.

```
YOLO26n summary (fused): 122 layers, 2,408,932 parameters, 0 gradients, 5.4 GFLOPs
```

for val result：

model | type | device | preprocess(ms) | inference(ms) | postprocess(ms)
-- | -- | -- | -- | -- | --
yolov8n | pth | CPU | 3.2 | 63.1 | 0.7
yolo11n | pth | CPU | 3.3 | 72.1 | 0.7
yolo26n | pth | CPU | 3.4 | 66.7 | 0.6
yolov8n | onnx | CPU | 31.7 | 77.5 | 0.3
yolo11n | onnx | CPU | 29.4 | 74.1 | 0.3
yolo26n | onnx | CPU | 29.7 | 65.3 | 0.3
yolov8n | pth | GPU | 2.2 | 4.9 | 2.6
yolo11n | pth | GPU | 1.9 | 6.5 | 2.1
yolo26n | pth | GPU | 2 | 7.4 | 1.7
yolov8n | onnx | GPU | 2.4 | 13 | 0.7
yolo11n | onnx | GPU | 2 | 9.9 | 0.6
yolo26n | onnx | GPU | 2.1 | 10.2 | 0.6

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tongchangD, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue routed quickly, and an Ultralytics engineer will also assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since this appears to be a benchmarking/performance question with comparative results, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) that includes the exact commands/script used for validation and export, the model sources used, input image size, batch size, ONNX runtime/provider details, and whether timing was measured with warmup/repeated runs. This will help us review the reported CPU speed and accuracy observations more reliably 🔍

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
What version of Ultralytics are you using? And what's the code you used to benchmark? Can you post output of `yolo checks`?

**Comment by tongchangD** [comment]
@Y-T-G 
use ultralytics==8.4.38
code is
```
# -*- encoding: utf-8 -*-
'''
@File    :   main   
@brief   :   
@License :   
@Modify Time      @Author    @Version    @Desciption
------------      -------    --------    -----------
2026/4/22 上午10:36   tongchangD      1.0         None
'''
import os
import torch
from ultralytics import YOLO
import ultralytics
print("ultralytics.__version__",ultralytics.__version__)
model_paths = [
"./yolo26n.pt",
]
for model_path_ in model_paths:
    model_path = model_path_
    for device in ["CPU","0"]:
        if os.path.exists(model_path):
            os.system("md5sum "+model_path)
            model = YOLO(model_path)
            model.fuse()
            metrics = model.val(data="./data.yaml", imgsz=640,task="detect", conf=0.25,iou=0.6, batch=6, device=device)
        else:
            print("not have path", model_path, "please check")
        torch.cuda.empty_cache()
```

**Comment by Y-T-G** [comment]
You can try comparing the other sizes of the model.

**Comment by Y-T-G** [comment]
How large is the dataset you're testing on?

**Comment by tongchangD** [comment]
@Y-T-G  A dataset comprising 3,000 images—each originally sized at 1024×1280 pixels—was uniformly resized to 640×640 pixels exclusively to facilitate computational speed evaluation during preliminary testing.
